standard/classes.md

```diff
@@ -5584,3 +5466,146 @@ When the body of the async function terminates, the return task is moved out of
+
+- If the state of the enumerator object is **before**, invoking `Dispose` changes the state to **after**.
+- If the state of the enumerator object is **running**, the result of invoking `Dispose` is unspecified.
+- If the state of the enumerator object is **suspended**, invoking `Dispose`:
+  - Changes the state to **running**.
+  - Executes any finally blocks as if the last executed `yield return` statement were a `yield break` statement. If this causes an exception to be thrown and propagated out of the iterator body, the state of the enumerator object is set to **after** and the exception is propagated to the caller of the `Dispose` method.
+  - Changes the state to **after**.
+- If the state of the enumerator object is **after**, invoking `Dispose` has no affect.
+
+### 15.14.6 Enumerable objects
+
+#### 15.14.6.1 General
+
+When a function member or local function returning an enumerable interface type is implemented using an iterator block, invoking the function member does not immediately execute the code in the iterator block. Instead, an ***enumerable object*** is created and returned.
+
+The enumerable object’s `GetEnumerator` or `GetAsyncEnumerator` method returns an enumerator object that encapsulates the code specified in the iterator block, and execution of the code in the iterator block occurs when the enumerator object’s `MoveNext` or `MoveNextAsync` method is invoked. An enumerable object has the following characteristics:
+
+- It implements `IEnumerable` and `IEnumerable<T>` or `IAsyncEnumerable<T>`, where `T` is the yield type of the iterator.
+- It is initialized with a copy of the argument values (if any) and instance value passed to the function member.
+
+An enumerable object is typically an instance of a compiler-generated enumerable class that encapsulates the code in the iterator block and implements the enumerable interfaces, but other methods of implementation are possible. If an enumerable class is generated by the compiler, that class will be nested, directly or indirectly, in the class containing the function member, it will have private accessibility, and it will have a name reserved for compiler use ([§6.4.3](lexical-structure.md#643-identifiers)).
+
+An enumerable object may implement more interfaces than those specified above.
+
+> *Note*: For example, an enumerable object may also implement `IEnumerator` and `IEnumerator<T>`, enabling it to serve as both an enumerable and an enumerator. Typically, such an implementation would return its own instance (to save allocations) from the first call to `GetEnumerator`. Subsequent invocations of `GetEnumerator`, if any, would return a new class instance, typically of the same class, so that calls to different enumerator instances will not affect each other. It cannot return the same instance even if the previous enumerator has already enumerated past the end of the sequence, since all future calls to an exhausted enumerator must throw exceptions. *end note*
```

Comment:
I don't think this last sentence is true. Earlier we have:

- If the state of the enumerator object is **after**, invoking `MoveNext` returns false.

(And that Dispose has no effect, and that Current is unspecified...)

Reply:
Removed that sentence in #672 as it's in a note anyway...